model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/authorization/StoreFactoryCacheSession.java

```diff
@@ -943,7 +998,17 @@ public void findByResource(String resourceId, String resourceServerId, Consumer<
         @Override
         public void findByResourceType(String resourceType, String resourceServerId, Consumer<Policy> consumer) {
             String cacheKey = getPolicyByResourceType(resourceType, resourceServerId);
-            cacheQuery(cacheKey, PolicyResourceListQuery.class, () -> getPolicyStoreDelegate().findByResourceType(resourceType, resourceServerId),
+            cacheQuery(cacheKey, PolicyResourceListQuery.class, () -> {
+                        List<Policy> policies = new ArrayList<>();
+                        getPolicyStoreDelegate().findByResourceType(resourceType, resourceServerId, new Consumer<Policy>() {
+                            @Override
+                            public void accept(Policy policy) {
+                                policies.add(policy);
+                                consumer.accept(policy);
```

Comment:
This was not necessary before, why do we need it now? `cacheQuery` method is calling `consumer.accept()`

Reply:
Otherwise, the query is not cached. I'm fixing this.